server/routes/events.js

```diff
@@ -12,218 +13,580 @@ router.get('/ping', function(req, res, next) {
+    .then(result => {
+      if(result.length) {
+        res.json(result);
+      } else {
+        res.status(404).json('No events where fb_id = ' + event_id + ' found.');
+      }
+    })
+    .catch(err => { return next(err) });
+});
+
+// Update user registration for a given event
+router.put('/:event_id/register/:user_id', function(req, res, next) {
+  const event_id = req.params.event_id;
+  const user_id = req.params.user_id;
+  const paid = req.body.paid;
+  if (paid == "true" || paid == "false") {
+    knex('event_registration').update({ has_paid: paid })
+      .where({ event_id: event_id, user_id: user_id })
+      .then(result => {
+        if (result) {
+          res.send(util.message('Successfully updated user registration for event with event_id ' + event_id));
+        } else {
+          util.throwError(404, 'No user found with user_id ' + user_id + ' for event with event_id ' + event_id);
+        }
+      })
+      .catch(err => { return next(err) });
+  } else {
+    util.throwError(400, 'Parameter paid is not a boolean')
+  }
 });
 
 // GET all users registered and have paid or not paid for a given event
 router.get('/:event_id/register', function(req, res, next) {
   const event_id = req.params.event_id;
   const paid = req.query.paid;
-  if (paid == undefined) {
-		db.query('SELECT user_id FROM event_registration WHERE event_id = $1', [event_id], (err, result) => {
-	  	if (err) return next(err);
-    	res.send(result.rows);
-  	});
-  } else if (paid == "true" || paid == "false") {
-		db.query('SELECT user_id FROM event_registration WHERE event_id = $1 AND has_paid = $2', [event_id, paid], (err, result) => {
-			if (err) return next(err);
-      res.send(result.rows);
-  	});
+
+  if (!paid || paid == "true" || paid == "false") {
+    if (!paid) {
+      knex('event_registration').select()
+        .where({ event_id: event_id })
+        .then(result => {
+          if(result.length) {
+            res.json(result);
+          } else {
+            res.status(404).json('No events with fb_id = ' + event_id + ' found.');
+          }
+        })
+        .catch(err => { return next(err) });
+    }
+    else {
+      knex('event_registration').select('user_id')
+        .where({ event_id: event_id, has_paid: paid })
+        .then(result => {
+         if(result.length) {
+           res.json(result);
+          } else {
+            res.status(404).json('No events with fb_id = ' + event_id + ' found.');
+          }
+        })
+        .catch(err => { return next(err) });
+    }
   } else {
-    return res.status(400).send({
-      message: '\'paid\' is not a boolean'
-    });
+    util.throwError(400, 'Parameter paid is not a boolean')
   }
 });
 
+// Register a user for a given event
+router.post('/:event_id/register/:user_id', function(req, res, next) {
+  const event_id = req.params.event_id;
+  const user_id = req.params.user_id;
+  const paid = req.body.paid;
+  if (!paid || paid != "true" || paid != "false") {
```

Comment:
This needs to be an and condition

lowercase paid first
if (paid != 'true' && paid != 'false') {
 then throw error
}

Reply:
Actually I don't think we require has_paid to be set? if it's not passed then an undefined value is fine. make the if statement:

if (paid != 'true' && paid != 'false') {
paid = undefined
}

this will make knex trigger the default value on insert